### ISSUE TITLE
macOS: crash on launch due to missing library "libuchardet.0.dylib"

### ISSUE THREAD

**Issue by mrbumpy409** [pedrolcl/dmidiplayer#26]
dmidiplayer 1.7.3 crashes on launch of both of my macOS installations:

* MacBook Pro, late 2013 model, running macOS Big Sur (11.7.10)
* VirtualBox install, running macOS Monterey (12.7.6)

Neither system can be upgraded beyond its current macOS version, the MacBook Pro due to being unsupported, and the VirtualBox install due to my host CPU's lack of AVX2 instructions (Intel Core i7-990x).

In both cases, the culprit appears to be a missing `libuchardet.0.dylib` library. Here are the crash logs:

* [dmidiplayer_crash_MBPro13_BigSur.txt](https://github.com/user-attachments/files/16737720/dmidiplayer_crash_MBPro13_BigSur.txt)
* [dmidiplayer_crash_VBox_Monterey.txt](https://github.com/user-attachments/files/16737721/dmidiplayer_crash_VBox_Monterey.txt)

**Comment by pedrolcl** [comment]
can you please execute in a terminal emulation this  command?

```
ls -la /Applications/dmidiplayer.app/Contents/Frameworks
```

it should list all libraries and frameworks bundled within the application, including libuchardet. If it is there, I don't know why is not loaded. Sorry.

Alternatively, you may try to install uchardet from homebrew: https://formulae.brew.sh/formula/uchardet
Homebrew has also a  cask providing dmidiplayer: https://formulae.brew.sh/cask/dmidiplayer

**Comment by mrbumpy409** [comment]
> can you please execute in a terminal emulation this command?
> 
> ```
> ls -la /Applications/dmidiplayer.app/Contents/Frameworks
> ```
> 
> it should list all libraries and frameworks bundled within the application, including libuchardet. If it is there, I don't know why is not loaded. Sorry.

Yes, I can see that `libuchardet.0.dylib` is indeed there. I have dmidiplayer installed through Homebrew, though I have the same issue when using it via the SourceForge download.

> Alternatively, you may try to install uchardet from homebrew: https://formulae.brew.sh/formula/uchardet

Aah, installing uchardet solved the issue. Thank you! The player now opens and plays MIDI files just fine, and the onboard FluidSynth is working as well.
